[yihfei] iP

yihfei opened the pull request:
# Haste
>"Success comes to people with haste"  -me

Hastes frees your mind of having to keep track of reminders. 

- text-based
- *easy* commands
- **lightweight**

All you need to do is
1. Download [here](https://github.com/yihfei/ip/releases/tag/A-Jar)
2. And get productive!


Features:
- [x] managing tasks
- [x] find tasks
- [x] deadlines
- [x] events
- [ ] auto save 

If you Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public class Main { 
    public static void main(String[] args) { 
        Application.launch(MainApp.class, args);
    }
 }
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

yihfei committed:
Level 0

yihfei committed:
Level 1, 2, 3

yihfei committed:
level 4

yihfei committed:
Level 4

yihfei committed:
A-TextUiTesting and minor formatting

yihfei committed:
Merge branch 'master' of github.com:yihfei/ip

yihfei committed:
Level 5

yihfei committed:
Level 6

yihfei committed:
Add TaskList

yihfei committed:
Add save functionality to TaskList

yihfei committed:
Change TaskList to handle missing data file

yihfei committed:
Merge level 7 branch with master branch

yihfei committed:
Add TimeParser class to handle Strings of date and time

yihfei committed:
Merge branch-level-8 to master

yihfei committed:
Add Ui and Parser classes to handle user interactions and commands respectively

yihfei committed:
Add Storage class and organise packages

yihfei committed:
Merge branch-oop to master

yihfei committed:
Organise packages

yihfei committed:
Merge branch-packages to master

yihfei committed:
Merge master to add-gradle-support

yihfei committed:
Add gradle with basic run functionality

yihfei committed:
Merge add-gradle-support to master

yihfei committed:
Organise packages into haste package

yihfei committed:
Fix minor bug

yihfei committed:
Change TaskList to be non-static and fix storage read and write

yihfei committed:
Add Command class

yihfei committed:
Finish Command class

yihfei committed:
Merge branch-command to master

yihfei committed:
Clean up code

yihfei committed:
Clean code and Add acknowledgements

yihfei committed:
Add StorageTest.java and ParserTest.java

yihfei committed:
Merge branch-junit to master

yihfei committed:
Enhance readability and fix indentations

yihfei committed:
Test

yihfei committed:
Test

yihfei committed:
Merge branch-codingstandard to master

yihfei committed:
Add JavaDoc comments

yihfei committed:
Add FindCommand.java and Add Javadoc comments

qz1004 commented on src/main/java/haste/commands/AddCommand.java:
```diff
@@ -0,0 +1,81 @@
+    private String desc;
+    private LocalDateTime start;
+    private LocalDateTime end;
+    private String type;
```
Perhaps you can use a more specific name for this variable? (eg. taskType)

yihfei replied:
Yeap I should have put a more descriptive name for variables like desc and type

qz1004 commented on src/main/java/haste/commands/AddCommand.java:
```diff
@@ -0,0 +1,81 @@
+     * @param end Deadline of the task.
+     * @param start Start time of the task.
+     */
+    public AddCommand(String desc, LocalDateTime end, LocalDateTime start) {
```
I like how you used method overloading here

yihfei replied:
Thank you!

qz1004 commented on src/main/java/haste/commands/AddCommand.java:
```diff
@@ -0,0 +1,81 @@
+    public void execute(TaskList tasks, Ui ui) {
+        Task newTask = null;
+        switch (this.type) {
+            case "t":
+                newTask = new ToDo(this.desc, false);
+                break;
+            case "d":
+                newTask = new Deadline(this.desc, this.end, false);
+                break;
+            case "e":
+                newTask = new Event(this.desc, this.start, this.end, false);
+                break;
```
According to the coding standards, case statements do not need to be indented

yihfei replied:
will rectify, thanks!

qz1004 commented on src/main/java/haste/commands/Parser.java:
```diff
@@ -0,0 +1,282 @@
+     * @param words The array of words user inputted.
+     * @return MarkCommand object containing index of task.
+     */
+    public static Command verifyUnmark(String[] words) {
```
Does this do the same thing as verifyMark? If so, I think this method can be omitted.

cbj252 replied:
It seems that verifyMark and verifyUnmark are identical with two exceptions - one returns a new MarkCommand while the other returns a UnmarkCommand and a small change in one of the texts.

@yihfei, maybe you could consider using a helper function that both verifyMark and verifyUnmark use?

qz1004 commented on src/main/java/haste/commands/Parser.java:
```diff
@@ -0,0 +1,282 @@
+     * @return AddCommand object containing description of task.
+     */
+    public static Command verifyTodo(String cmd) {
+        Pattern pt = Pattern.compile("todo( (.+))?"); // ( ...)? is optional group
+        Matcher mt = pt.matcher(cmd);
```
pt and mt can be quite ambiguous as variable names

yihfei replied:
yeap, will change it

qz1004 commented on src/main/java/haste/commands/Parser.java:
```diff
@@ -0,0 +1,282 @@
+     * @param savedTime LocalDateTime object stored in task.
+     * @return String representation of time.
+     */
+    public static String getCmd(LocalDateTime savedTime) {
```
Perhaps you can consider getStringFormat or getCmdFormat as a method name for better clarity?

yihfei replied:
Thanks for the suggestion! will rectify

qz1004 commented on src/main/java/haste/data/Storage.java:
```diff
@@ -0,0 +1,136 @@
+     * @param line String representation of task from data file.
+     * @return Task object.
+     */
+    public Task readTask(String line) {
```
Perhaps you can consider using getTask or toTask as a method name to avoid confusion with the read method

yihfei replied:
Alright, will take your suggestion into account!

qz1004 commented on src/main/java/haste/tasks/Task.java:
```diff
@@ -0,0 +1,55 @@
+     *
+     * @return String representation of Task.
+     */
+    public String save() {
```
save was already used as a method name in another class. Perhaps you can consider using something along the lines of toSaveFormat or toFileFormat instead

yihfei replied:
Alright, will take your suggestion into account!

qz1004 commented on src/main/java/haste/ui/Ui.java:
```diff
@@ -0,0 +1,60 @@
+        this.running = false;
+    }
+
+    public void mark(String taskDesc, TaskList tasks) {
```
I think you can consider using method names that are more specific to this class, eg. addMessage, markMessage etc. to avoid confusion with methods in other classes

yihfei replied:
Alright, will take your suggestion into account!

qz1004 reviewed (commented):
Overall, your code was easy to understand and I like your use of packages and method overloading. Some of the method/variable names can be tweaked to make your code clearer.

yihfei committed:
Add GUI

cbj252 commented on src/main/java/haste/commands/Parser.java:
```diff
@@ -0,0 +1,282 @@
+     * @return MarkCommand object containing index of task.
+     */
+    public static Command verifyMark(String[] words) {
+        int id ;
```
Extra whitespace.

cbj252 commented on src/main/java/haste/tasks/Deadline.java:
```diff
@@ -0,0 +1,34 @@
+    public Deadline(String description, LocalDateTime by, boolean isComplete) {
+        super(description,isComplete);
+        this.by = by;
+    }
```
Logical units within a block should be separated by one blank line - there's no line inbetween here and @Override on line 25.

cbj252 commented on src/main/java/haste/tasks/Event.java:
```diff
@@ -0,0 +1,35 @@
+     * @param isComplete Completion status of Task.
+     */
+    public Event(String description, LocalDateTime start, LocalDateTime end, boolean isComplete) {
+        super(description,isComplete);
```
White space within a statement - Java reserved words should be followed by a white space.

cbj252 commented on src/main/java/haste/data/Storage.java:
```diff
@@ -0,0 +1,136 @@
+            e.printStackTrace();
+        }
+    }
+    // add one line to text file
```
Not a coding standard violation - but confusing to read. What does this line (// add one line to text file) refer to?

cbj252 commented on src/main/java/haste/data/TaskList.java:
```diff
@@ -0,0 +1,78 @@
+ * Represents a list of Task objects.
+ */
+public class TaskList {
+    public ArrayList<Task> taskList;
```
Class variables should never be declared public: Here, taskList already has all the functions necessary that both taskList and numOfTasks can be made private w/o any other changes.

cbj252 reviewed (commented):
Overall, I like the use of the commands package and inheritance, making your code easy to read.

yihfei committed:
Change GUI to respond to command

yihfei committed:
Merge branch-level-10 to master

yihfei committed:
Clean up code and add greeting when initialized

yihfei committed:
Clean up code using checkstyle

yihfei committed:
Merge branch-checkstyle to master

yihfei committed:
Update build.gradle for JAR file

Updated GUI for Haste chatbot which changes the main classes of chatbot.

Modify main class in build.grade to Launcher.java to launch the GUI in Jar file.

yihfei committed:
Add assert statements

Many methods require assumptions about certain variables.

Use assert statement to ensure that these assumptions hold true.

yihfei committed:
Remove assert false statements

yihfei committed:
Merge pull request #1 from yihfei/branch-Assertions

Branch assertions

yihfei committed:
Revert "Branch assertions"

yihfei committed:
Merge pull request #2 from yihfei/revert-1-branch-Assertions

Revert "Branch assertions"

yihfei committed:
Revert "Remove assert false statements"

This reverts commit d9dfa135210e2a4a581367fd8ab97fc5c4d75217.

Remove latest commit

yihfei committed:
Change ui method called in UmarkCommand

yihfei committed:
Merge pull request #3 from yihfei/branch-Assertions

Branch assertions

yihfei committed:
Minor changes

yihfei committed:
Trial

yihfei committed:
Trial

yihfei committed:
Add assert statements

Methods in haste package require certain assumptions.

Assert makes sure that these assumption holds true.

yihfei committed:
Format change

yihfei committed:
Merge branch 'master' into branch-Assertions

yihfei committed:
Merge pull request #4 from yihfei/branch-Assertions

Branch assertions

Fix problem brought by revert pull request

yihfei committed:
Fix merge conflict issue

yihfei committed:
Remove duplicate method

Goes against DRY coding principle.

Keep one method that does the functionality of both and remove the other.

Keep code concise and remove unnecessary code.

yihfei committed:
Merge pull request #5 from yihfei/branch-CodeQuality

Remove duplicate method

yihfei committed:
Improve code quality

Some methods look complicated with big chunks of code.

Add lines between logic blocks and remove unused variables and imports.

Improves readability and reduces confusion over unimportant code.

yihfei committed:
Merge branch 'C-BetterSearch'
Add BetterSearch functionality

Haste find command requires full keyword to be present.

Modify find to take partial keyword to find specific tasks.

Find command would be able to locate certain tasks without the whole keyword keyed in.

yihfei committed:
Add test case to StorageTest

yihfei committed:
Update Parser.java

yihfei committed:
Add Ui.png and user guide to README.md

yihfei committed:
Remove commented out code

yihfei committed:
Update README.md
